// devextreme/esm/localization/cldr-data/first_day_of_week_data.js
/**
 * DevExtreme (esm/localization/cldr-data/first_day_of_week_data.js)
 * Version: 21.2.15
 * Build date: Fri Dec 08 2023
 *
 * Copyright (c) 2012 - 2023 Developer Express Inc. ALL RIGHTS RESERVED
 * Read about DevExtreme licensing here: https://js.devexpress.com/Licensing/
 */
// !!! AUTO-GENERATED FILE, DO NOT EDIT
export default {
    "aa-DJ": 6,
    "aa-ER": 1,
    ab: 1,
    "af-NA": 1,
    agq: 1,
    ak: 1,
    an: 1,
    ann: 1,
    apc: 6,
    ar: 6,
    "ar-EH": 1,
    "ar-ER": 1,
    "ar-KM": 1,
    "ar-LB": 1,
    "ar-MA": 1,
    "ar-MR": 1,
    "ar-PS": 1,
    "ar-SO": 1,
    "ar-SS": 1,
    "ar-TD": 1,
    "ar-TN": 1,
    arn: 1,
    asa: 1,
    ast: 1,
    az: 1,
    "az-Arab": 6,
    "az-Arab-TR": 1,
    "az-Cyrl": 1,
    ba: 1,
    bas: 1,
    be: 1,
    bem: 1,
    bez: 1,
    bg: 1,
    "bgn-AE": 6,
    "bgn-AF": 6,
    "bgn-IR": 6,
    "bgn-OM": 6,
    blo: 1,
    blt: 1,
    bm: 1,
    "bm-Nkoo": 1,
    bo: 1,
    br: 1,
    bs: 1,
    "bs-Cyrl": 1,
    bss: 1,
    byn: 1,
    ca: 1,
    cch: 1,
    ce: 1,
    cgg: 1,
    ckb: 6,
    co: 1,
    cs: 1,
    cu: 1,
    cv: 1,
    cy: 1,
    da: 1,
    de: 1,
    dje: 1,
    dsb: 1,
    dua: 1,
    dv: 5,
    dyo: 1,
    ee: 1,
    el: 1,
    "en-001": 1,
    "en-AE": 6,
    "en-BI": 1,
    "en-MP": 1,
    "en-MV": 5,
    "en-SD": 6,
    "en-Shaw": 1,
    eo: 1,
    es: 1,
    et: 1,
    eu: 1,
    ewo: 1,
    fa: 6,
    ff: 1,
    "ff-Adlm": 1,
    fi: 1,
    fo: 1,
    fr: 1,
    "fr-DJ": 6,
    "fr-DZ": 6,
    "fr-SY": 6,
    frr: 1,
    fur: 1,
    fy: 1,
    ga: 1,
    gaa: 1,
    gd: 1,
    "gez-ER": 1,
    gl: 1,
    gsw: 1,
    gv: 1,
    ha: 1,
    "ha-Arab": 1,
    "ha-Arab-SD": 6,
    hr: 1,
    hsb: 1,
    hu: 1,
    hy: 1,
    ia: 1,
    ie: 1,
    ig: 1,
    ii: 1,
    io: 1,
    is: 1,
    it: 1,
    jbo: 1,
    jgo: 1,
    jmc: 1,
    ka: 1,
    kab: 6,
    kaj: 1,
    kcg: 1,
    kde: 1,
    kea: 1,
    ken: 1,
    khq: 1,
    kk: 1,
    kkj: 1,
    kl: 1,
    "ko-CN": 1,
    "ko-KP": 1,
    kpe: 1,
    ksb: 1,
    ksf: 1,
    ksh: 1,
    ku: 1,
    kw: 1,
    ky: 1,
    la: 1,
    lag: 1,
    lb: 1,
    lg: 1,
    lij: 1,
    lmo: 1,
    ln: 1,
    lrc: 6,
    lt: 1,
    lu: 1,
    lv: 1,
    "mas-TZ": 1,
    mdf: 1,
    mfe: 1,
    mg: 1,
    mgo: 1,
    mi: 1,
    mk: 1,
    mn: 1,
    "mn-Mong": 1,
    ms: 1,
    "ms-Arab": 1,
    mua: 1,
    myv: 1,
    mzn: 6,
    naq: 1,
    nds: 1,
    nl: 1,
    nmg: 1,
    nnh: 1,
    no: 1,
    nqo: 1,
    nus: 1,
    ny: 1,
    nyn: 1,
    oc: 1,
    os: 1,
    pap: 1,
    pcm: 1,
    pis: 1,
    pl: 1,
    prg: 1,
    ps: 6,
    "pt-AO": 1,
    "pt-CH": 1,
    "pt-CV": 1,
    "pt-GQ": 1,
    "pt-GW": 1,
    "pt-LU": 1,
    "pt-ST": 1,
    "pt-TL": 1,
    "qu-BO": 1,
    "qu-EC": 1,
    rif: 1,
    rm: 1,
    rn: 1,
    ro: 1,
    rof: 1,
    ru: 1,
    rw: 1,
    rwk: 1,
    sah: 1,
    sbp: 1,
    sc: 1,
    scn: 1,
    sdh: 6,
    se: 1,
    ses: 1,
    sg: 1,
    shi: 1,
    "shi-Latn": 1,
    si: 1,
    sk: 1,
    sl: 1,
    sma: 1,
    smj: 1,
    smn: 1,
    sms: 1,
    so: 1,
    "so-DJ": 6,
    sq: 1,
    sr: 1,
    "sr-Latn": 1,
    "ss-SZ": 1,
    ssy: 1,
    "st-LS": 1,
    sv: 1,
    sw: 1,
    syr: 6,
    szl: 1,
    "ta-LK": 1,
    "ta-MY": 1,
    teo: 1,
    tg: 1,
    "ti-ER": 1,
    tig: 1,
    tk: 1,
    to: 1,
    tok: 1,
    tpi: 1,
    tr: 1,
    tt: 1,
    twq: 1,
    tyv: 1,
    tzm: 1,
    ug: 1,
    uk: 1,
    uz: 1,
    "uz-Arab": 6,
    "uz-Cyrl": 1,
    vai: 1,
    "vai-Latn": 1,
    vec: 1,
    vi: 1,
    vo: 1,
    vun: 1,
    wa: 1,
    wae: 1,
    wbp: 1,
    wo: 1,
    xog: 1,
    yav: 1,
    yi: 1,
    yo: 1,
    "yue-Hans": 1,
    za: 1,
    zgh: 1,
    zh: 1
};
